client/src/components/prospects/Table.jsx

```diff
@@ -24,13 +24,46 @@ export default function ProspectsTable(props) {
 
   const [rows, setRows] = useState([]);
   const [filteredRows, setFilteredRows] = useState([]);
+  const [buttonDisabled, setButtonDisabled] = React.useState(true);
 
   function fetchProspects() {
     Auth.getProspects().then((res) => {
-      setRows(res.data.prospects);
-      setFilteredRows(res.data.prospects);
+      let pList = res.data.prospects;
+      pList.map(p => {p['checked'] = false});
+
+      setRows(pList);
+      setFilteredRows(pList);
     });
+  }
 
+  function handleCheck(isChecked, id) {
+    let disabled = buttonDisabled;
+    // 'rows' is read-only, so we need to copy everything and then change.
+    // See https://stackoverflow.com/a/49502115
+    for (let i=0; i < rows.length; i++) {
+      if (rows[i]['id'] === id) {
+        // 1. Make a shallow copy of the items
+        let items = [...rows];
```

Comment:
Wouldn't it make more sense to make a shallow of of the rows array outside of your for loop so you only need to do it once?

Reply:
It should only happen once though per call because handleCheck only happens for a single id, so the if-statement should only apply to that single id.